### PR TITLE
Add serialization for universal polynomials

### PR DESCRIPTION
**Pull request by SoongNoonien**
Working with the universal polynomials I noticed that they can't be serialized. This should be fixed now.

### CODE REVIEW

File: src/Serialization/Rings.jl [hunk 1]
```diff
@@ -60,35 +62,40 @@ end
...
     end
 
     return polynomial_ring(base_ring, symbols, cached=false)[1]
 end
 
 ################################################################################
-# Multivariate Polynomials
+# Multivariate and Universal Polynomials
 @registerSerializationType(QQMPolyRingElem)
 @registerSerializationType(ZZMPolyRingElem)
```
Comment (fingolfin) [inline src/Serialization/Rings.jl:92]: This is *not a review of this PR* but rather a separate question for @antonydellavecchia : I am actually surprised that we list `QQMPolyRingElem`, `ZZMPolyRingElem`, etc. separately here. Shouldn't these all be serialized as `MPolyRingElem`, with the `parent` and or `base_ring` determining the full type?

**Review by fingolfin — APPROVED** [review]
Thank you! This looks good to me, but it would be good if @antonydellavecchia had a look, too, before we merge this

File: src/Serialization/Rings.jl (hunk 1 above)
Comment (antonydellavecchia) [reply src/Serialization/Rings.jl:92]: it does seem like these are no longer needed, I'll make a seperate pull requests and get rid of some unnecessary registration

**Comment by antonydellavecchia** [comment]
also looks good to me

**Pull request merged**